bodhi/tests/server/consumers/test_masher.py

```diff
@@ -1231,3 +1235,69 @@ def test_comps_unsafe_http_url(self, mock_cmd):
+                'f23-updates-161003.1302', 'f23-updates-161004.1423', 'f23-updates-161005.0259',
+                'f23-updates-testing-161001.0424', 'f23-updates-testing-161003.0621',
+                'f23-updates-testing-161003.2217', 'f24-updates-161002.2331',
+                'f24-updates-161003.1302', 'f24-updates-testing-161001.0424',
+                'this_should_get_left_alone', 'f23-updates-should_be_untouched',
+                'f23-updates.repocache', 'f23-updates-testing-blank']
+            [os.makedirs(os.path.join(mash_dir, d)) for d in dirs]
+            # Now let's make a few files here and there.
+            with open(os.path.join(mash_dir, 'dist-5E-epel-161003.0724', 'oops.txt'), 'w') as oops:
+                oops.write('This mash failed to get cleaned and left this file around, oops!')
+            with open(os.path.join(mash_dir, 'COOL_FILE.txt'), 'w') as cool_file:
+                cool_file.write('This file should be allowed to hang out here because it\'s cool.')
+
+            with patch.dict(config.config, {'mash_dir': mash_dir}):
+                result = testing.CliRunner().invoke(masher.clean_up, [])
+
+            self.assertEqual(result.exit_code, 0)
+            # We expect these and only these directories to remain.
+            expected_dirs = {
+                'dist-5E-epel-161012.1854', 'dist-5E-epel-161013.1711',
+                'dist-5E-epel-testing-161003.0856', 'dist-5E-epel-testing-161006.0053',
+                'dist-6E-epel-161002.2331', 'dist-6E-epel-161003.2046',
+                'dist-6E-epel-testing-161001.0528', 'epel7-161004.1423', 'epel7-161005.1122',
+                'epel7-testing-161003.0621', 'epel7-testing-161003.2217', 'f23-updates-161004.1423',
+                'f23-updates-161005.0259', 'f23-updates-testing-161003.0621',
+                'f23-updates-testing-161003.2217', 'f24-updates-161002.2331',
+                'f24-updates-161003.1302', 'f24-updates-testing-161001.0424',
+                'this_should_get_left_alone', 'f23-updates-should_be_untouched',
+                'f23-updates.repocache', 'f23-updates-testing-blank'}
+            actual_dirs = set([d for d in os.listdir(mash_dir)
+                               if os.path.isdir(os.path.join(mash_dir, d))])
+            self.assertEqual(actual_dirs, expected_dirs)
+            # The cool file should still be here
+            actual_files = [f for f in os.listdir(mash_dir)
+                            if os.path.isfile(os.path.join(mash_dir, f))]
+            self.assertEqual(actual_files, ['COOL_FILE.txt'])
+            # Make sure the printed output is correct
+            expected_output = set(dirs) - expected_dirs
+            expected_output = {os.path.join(mash_dir, d) for d in expected_output}
+            expected_output = expected_output | {'Deleting the following directories:', ''}
+            self.assertEqual(set(result.output.split('\n')), expected_output)
+        finally:
+            shutil.rmtree(mash_dir)
```

Comment:
This test is copied from the existing test. We don't need to retest that this code works, but we do need to assert that the masher calls it when done.